### ISSUE TITLE
Library: Add some context about the currently selected category to the main 'Library Content' region

### ISSUE THREAD

**Issue by talldan** [WordPress/gutenberg#52006]
### Description

In the library,  a user selects a category using the options in the 'navigation' region.
![Screen Shot 2023-06-28 at 11 44 52 am](https://github.com/WordPress/gutenberg/assets/677833/494c5787-1b74-4800-b6ea-52ef7f3013cc)

In the main 'Library content' region there's no contextual clues as to what category is currently being viewed.
![Screen Shot 2023-06-28 at 11 46 01 am](https://github.com/WordPress/gutenberg/assets/677833/1e9b623a-854d-49ce-86e5-4bd41cbe069a)

This would likely be frustrating for anyone using a screenreader, as the only way to know the selected category would be to check back through the navigation items to find the one currently selected.

To improve this, I think it'd be good to add a heading at the start of the main area. If it doesn't fit with the design it could be `VisuallyHidden`.

### Step-by-step reproduction instructions

1. Open the site editor
2. Tab to the Library option in the Navigation region and select it
3. Tab through the available categories that appear and select one
3. Use the 'navigate regions' shortcut to move to the 'Library Content' region
4. Search through the content in that region and observe there's no indication of the selected category.

### Please confirm that you have searched existing issues in the repo.

Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

Yes

**Comment by joedolson** [comment]
I think this would be a good improvement; having some context clues about what kind of content you're currently working with at a high and granular level is very valuable. 

I'll note that there is a heading in place - 'Synced/Standard' is an H4. This is an invalid heading sequence; given the context, it should be an H2. 

I don't see any heading that indicates the context being Patterns, let alone categories of patterns; all of that would be useful.

For any given category, there should be a heading at level H2 to give context. With that, the current H4 should move to H3 so that the hierarchy is consistent. 

I'd prefer a visible heading, particularly on mobile, where the sidebar isn't visible while working in the library.

**Comment by jameskoster** [comment]
I think recent updates to the pattern library mean this can be closed. Sections all now have a contextual title and the majority also have a description. Some examples:

<img width="1358" alt="Screenshot 2023-07-10 at 21 39 02" src="https://github.com/WordPress/gutenberg/assets/846565/641b22f7-cdaa-4b29-8f14-4d6ad382a9d4">
<img width="1359" alt="Screenshot 2023-07-10 at 21 39 20" src="https://github.com/WordPress/gutenberg/assets/846565/e1ee874b-509a-4807-b85c-4dc2dd00a9af">
<img width="1358" alt="Screenshot 2023-07-10 at 21 39 32" src="https://github.com/WordPress/gutenberg/assets/846565/8765af2b-ea7b-4022-8cda-34e8645f919b">
